src/snowflake/snowpark/async_job.py

```diff
@@ -284,10 +284,28 @@ def cancel(self) -> None:
                 "ENABLE_ASYNC_QUERY_IN_PYTHON_STORED_PROCS", False
             )
         ):
-            cancel_resp = self._session._conn._conn.cancel_query(self.query_id)
-            if not cancel_resp.get("success", False):
+            import _snowflake
+            import json
+
+            raw_cancel_resp = _snowflake.cancel_query(self.query_id)
```

Comment:
Another way is to just do `self._cursor.execute(f"select SYSTEM$CANCEL_QUERY('{self.query_id}')")`. It seems cleaner, but a different behavior comparing to calling the cancel query API.